posts/wordpress-to-jamstack.md

```diff
@@ -0,0 +1,152 @@
+
+Since I am using serverless functions and NodeJS I had to take a different approach. I ended up signing up for an account on SendGrid because they have a very generous free tier and using [nodemailer](https://nodemailer.com/about/). Once I signed up on SendGrid and got the smtp settings it was pretty straightforward. The contact form calls the api I created and then that creates the email based on the template I defined and sends it to the Church's inbox.
+
+## Mail API
+
+```javascript
+const nodemailer = require("nodemailer");
+
+// async..await is not allowed in global scope, must use a wrapper
+export default async (_req, res) => {
+  const req = _req.body;
+
+  // create reusable transporter object using the default SMTP transport
+  const transporter = nodemailer.createTransport({
+    host: "smtp.sendgrid.net",
+    port: 587,
+    secure: false, // true for 465, false for other ports
+    auth: {
+      user: "apikey",
+      pass: process.env.SMTP_API,
+    },
+  });
+
+  // send mail with defined transport object
+  const info = await transporter.sendMail({
+    from: process.env.SMTP_FROM, // sender address
+    to: process.env.SMTP_TO, // list of receivers
+    subject: req.subject, // Subject line
+    html: `<p><strong>Name:</strong> ${req.name}</p><p><strong>Email:</strong> ${req.email}</p><p><strong>Message:</strong> ${req.message}</p>`, // html body
+  });
+
+  if (info.response.includes("250 Ok")) {
+    res.status(200);
+    res.send(info.messageId);
+  } else {
+    res.status(500);
+    res.send("Error Sending Message");
+  }
+};
+```
+
+# Conclusion
+
+Overall this was a pretty fun and sometimes challenging side project. I was able to explore some new technologies while also saving my parents church some yearly expenses by getting rid of their old Wordpress site. During all of this they also decided to do some re-branding and got a new logo for the church and color scheme. It was fun to be apart of this transformation and see everything come together. My biggest concern when doing this was not to accrue any expenses for the church if at all possible. I was able to accomplish this because most of the services I chose offer free tiers that this website fit in or offered credits like DataStax does. So instead of paying someone around $200/year to maintain their old website they will just now have to pay for their domain renewal every couple years.
```

Comment:
```suggestion
Overall this was a pretty fun and at times challenging side project. I was able to explore new technologies while also saving my parents' church yearly expenses by replacing their old Wordpress site. During all of this they also felt they could enhance their brand by updating their church logo and color scheme. It was fun to be part of this transformation and see everything come together. My biggest concern when doing this was not to accrue any expenses for the church if at all possible. I was able to accomplish this because most of the services I chose offer free tiers (that this particular use case fell in) or offered credits like DataStax does. So instead of paying maintenance fees of about $200/year for their old website they will just now have to pay for their domain renewal every couple years.
```

Reply:
maybe add how much domain renewal is too?

Reply:
Also might want to add a little snippet at the end that is – check out the screen shots below for the before & after!